### PR TITLE
Canisters of Holding

### PR DESCRIPTION
**Pull request by Soup-code**
## About The Pull Request
Adds Canisters of Holding that hold up to 3000L of gas. You can build them with Canister of Holding Parts printed at Engineering Protolathes after researching Miniaturized Bluespace.

Annotation 2019-06-13 133802

## Why It's Good For The Game
I like canisters and maybe emergent gameplay?

## Changelog
:cl: Soup-code
add: Added canisters of holding.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
what could possibly  go wrong

**Comment by zxaber** [comment]
We already have a max cap in place, it'll be _fine_.

**Comment by zxaber** [comment]
Also, if you weld one that burst, do you get the canister of holding parts back?

**Comment by wesoda25** [comment]
I don't think this will go very well. 

But its also not the most imaginative sprite, is it?

**Comment by knacker48** [comment]
I'm not good at atmos. Does this break anything?

**Comment by ExcessiveUseOfCobblestone** [comment]
why is it good for the game? What reasonable usecase is there for going past the current volume?

**Comment by ishitbyabullet** [comment]
@ExcessiveUseOfCobblestone well you see cobby sometimes your sm engine setup just won't work with 2 canisters of miasma so you gotta plop it all down into one

In all seriousness, I guess it's a longer lifespan for people who use their jetpacks heavily for space exploration.

**Comment by AdamElTablawy** [comment]
this is going to get ignored for like a year until an elder atmosian god uses one of these to somehow break the laws of thermodynamics and replace space with fire, i'm calling it now

**Comment by habader** [comment]
dont merge this jesus christ

**Comment by zxaber** [comment]
Cobby, I could see me using this to collect Pluox from a CO2 engine without having to swap cans every five minutes.

Though honestly, the real engineering solution is to figure out which starting atmos tank you don't need, vent it to space, and then just dump all the Pluox into the waste network to be recovered in the now-vacant tank.

**Comment by Whoneedspacee** [comment]
prefer pumps anyways this’ll probably be for fusion and nothing else. it’s no different storage wise than cooling the temperatue by 2.5x or so

**Comment by nemvar** [comment]
I guess it's kinda useful for harvesting air from lavaland. I always thought that it was weird that triple layered pipes had more volume than a canister.

**Comment by datorangebottle** [comment]
> Though honestly, the real engineering solution is to figure out which starting atmos tank you don't need, vent it to space, and then just dump all the Pluox into the waste network to be recovered in the now-vacant tank.

How about the tank that starts empty? Nobody uses the mix tank for anything legitimate anyway.

**Comment by zxaber** [comment]
That tank seems to be meant as a catch-all for anything not otherwise filtered into a dedicated tank. For times you want to save that Trit that someone released into the halls, I guess.

**Comment by Barhandar** [comment]
>Cobby, I could see me using this to collect Pluox from a CO2 engine without having to swap cans every five minutes.

And this will just mean swapping cans every 15 minutes, but way later in the round. Also, just cool your plox, it'll fit everything you ever need in a single can.

**Comment by Whoneedspacee** [comment]
Yeah I'm honestly against adding this just because of how pointless it is to be honest.

**Comment by optimumtact** [comment]
none of the peopel freaking out about this ahve even indicated something broken that they could even do with it

**Comment by HomocidlePancakes** [comment]
! C O D E R S P R I T E S!

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -23,6 +23,7 @@
 	pressure_resistance = 7 * ONE_ATMOSPHERE
 	var/temperature_resistance = 1000 + T0C
 	var/starter_temp
+	var/bluespace = FALSE //is it a canister of holding?
```
Comment (duncathan) [inline code/modules/atmospherics/machinery/portable/canister.dm]: you should probably override update icon in the subtype instead of using a var like this

**Review by duncathan — CHANGES_REQUESTED** [review]

**Comment by duncathan** [comment]
this would be broken if we didn't have a bombcap but all this means is peeps can pump more gas into one canister which at worst means like, canister floods are better? idk it can't be THAT bad, despite sounding p frightening at first

**Comment by CPTANT** [comment]
cannister floods are extremely tame compared to flooding the distro system from atmos.

**Comment by Unit2E** [comment]
People, canisters have nothing to do with bombs. People screaming "oh no bombs" are flat out wrong, heck, the opposite is true, canisters that hold more volume make canister bombs need more ingredients. Bombs (of the toxins kind) are entirely contained within tank code and larger canisters don't do anything for them. Even for flooding gas with a canister, playing with freezers allows you to easily fit a ton of gas even as it is now. This canister won't break anything, chill.

All this can reasonably be used for is simply storing more gas in a can, which is not as relevant as you think for most applications because freezers exist. This will make gas loss less of an issue doing certain atmos operations, though, which is nice.

**Comment by necromanceranne** [comment]
What if I fill it with lots and lots and lots of supercooled water vapour?

**Comment by Unit2E** [comment]
Oh for sure, it's mostly for memes that aren't very practical. As it is now, a 1000L can completely filled with 2.7K water vapour is a pain to make, it'll take you ages, assuming you don't run out of oxygen first. Other gas floods may very be somewhat more potent, if there is enough gas of it (just the base gasses, really). 

It does make the process slightly faster, too, as the volume pump and freezer won't have to struggle as much, even still, there's hardly enough gas on the entire station to entirely need the extra capacity, what would you use it for? In a normal can, using a volume pump and a freezer at 2.7K, you can fill it up to (9000*1000)/(8.31*2.7) = 401123 moles, on the LOW end. A can with 3000L volume will fit roughly 1203369 moles, again, somewhat low end. That's about 26 times the amount of gas of N2O on the station, in total. 2 times the amount of total plasma, and coming close to 3x the amount of CO2. Have fun flooding cold plasma or CO2 though, I guess!

**Comment by 0d0be32b607ee4ecbc2358f3dd4a27de** [comment]
do it. feed the fire.

**Comment by Jalleo** [comment]
Are we certain a full bluespace can cant get more moles of oxygen in a internals tank?

**Comment by Unit2E** [comment]
Yes, so long as the release pressure isn't changed, but looking at it, that didn't change. Nevermind that air pumps are already way better for this purpose

**Comment by Whoneedspacee** [comment]
why these are useless:

if you cool something 3x as much, it can store 3x the gas

a canister at 3 kelvin can store as much as a bluespace canister at 9 kelvin.

canisters work off of release pressure and can only go up to 1013 kpa or their own pressure. so they can only possibly put 1013kpa into a tank.

air pumps already exist and more efficiently pump 2533 kpa in instead of just releasing it in. if the pressure of the canister is less than the amount you want it can only release up to its pressure. rather than being pumped in its simply opened up and the pressure of the air in the canister moves into the tank.

there’s nothing you can do with this canister that you can’t do with something else, and usually easier.

it’s actually less efficient hooked up to a freezer loop because it takes up so much of the volume the freezer is cooling such a minimal amount of the mix.

its only real use is an early game canister fusion speedrun which was made obsolete anyways with the new fusion update favoring burns doing fusion, so its actually fairly worthless all things considered.

**Comment by Kollock5** [comment]
the only real use for the tank is for the atmos clown that wants to make the hole station slippery but doesn't want to freeze everybody to death...
HONK i like it

**Comment by optimumtact** [comment]
this guy is gitbanned for abusing the code review system to make undeletable comments

**Comment by timothymtorres** [comment]
Why would undeletable comments be a problem?  And I’m assuming it wasn’t done accidentally as well?

**Comment by optimumtact** [comment]
they're a problem because we can't delete them

**Comment by CPTANT** [comment]
Does it really change whether or not this PR is a good idea?

**Comment by optimumtact** [comment]
no but the OP isn't around to keep working on it, if you want to have a go then knock yourself out

**Pull request closed without merging**